File: src/main/java/it/contrader/model/Eventi.java
package it.contrader.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Inheritance(strategy = InheritanceType.JOINED)
@OnDelete(action = OnDeleteAction.CASCADE)
public class Eventi {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer idEvento;
	
	@Column(unique = true)
	private String nameevento;
	
	private String meseevento;
	
	@ManyToOne
	@JoinColumn(name="idCity")
	private City city;

	@ManyToOne
	@JoinColumn(name="idUser")
	private User user;

}

File: src/app/models/Eventi.ts
import { City } from './City';
import { User } from './User';
export class Eventi {
    idEvento: number;
    nameevento: string;
    meseevento: string;
    cityDTO: City;
    userDTO: User;

    // tslint:disable-next-line:max-line-length
    constructor(idEvento: number, nameevento: string, meseevento: string, cityDTO: City,  userDTO: User) {
        this.idEvento = idEvento;
        this.nameevento = nameevento;
        this.meseevento = meseevento;
        this.cityDTO = this.cityDTO;
        this.userDTO = this.userDTO ;
    }

}
